### PR TITLE
[Mac] Add filtering capabilities to brush popUp controller.

### PR DESCRIPTION
**Pull request by CartBlanche**
Video of filtering in action - 
https://www.screencast.com/t/Nc99Xdpfd

### CODE REVIEW

File: Xamarin.PropertyEditing.Mac/Controls/Custom/BrushTabViewController.cs [hunk 1]
```diff
@@ -10,16 +10,20 @@ namespace Xamarin.PropertyEditing.Mac
...
+		private NSSearchField filterResource;
+
 		public BrushTabViewController ()
 		{
 			PreferredContentSize = new CGSize (430, 230);
 			TransitionOptions = NSViewControllerTransitionOptions.None;
 			EdgeInsets = new NSEdgeInsets (0, 12, 12, 12);
 		}
 
-		Dictionary<CommonBrushType, int> BrushTypeTable = new Dictionary<CommonBrushType, int> ();
```
Comment (ermau) [inline Xamarin.PropertyEditing.Mac/Controls/Custom/BrushTabViewController.cs]: Why move them if you're not going to move them to the right spot?

**Comment by ermau** [comment]
My mentioned changes have been merged (see #417 )

**Comment by CartBlanche** [comment]
@ermau Rebased.

**Comment by ermau** [comment]
This has an issue where occasionally filtering will flash previews to the wrong color and back again.

**Comment by CartBlanche** [comment]
@ermau Please give this another try.

**Pull request merged**